mlflow/exceptions.py

```diff
@@ -119,7 +119,10 @@ class RestException(MlflowException):
     def __init__(self, json):
         self.json = json
 
-        error_code = json.get("error_code", ErrorCode.Name(INTERNAL_ERROR))
+        if not isinstance(json, dict):
+            error_code = ErrorCode.Name(INTERNAL_ERROR)
+        else:
+            error_code = json.get("error_code", ErrorCode.Name(INTERNAL_ERROR))
```

Comment:
```suggestion
        if isinstance(json, dict):
            error_code = json.get("error_code", ErrorCode.Name(INTERNAL_ERROR))
        else:
            error_code = ErrorCode.Name(INTERNAL_ERROR)
```

can we swap the conditions?